ledger/src/shred.rs

```diff
@@ -551,6 +551,10 @@ impl Shred {
             Self::ShredData(_) => Err(Error::InvalidShredType),
         }
     }
+
+    pub fn merkle_root(&self) -> Option<Hash> {
+        layout::get_merkle_root(self.payload())
+    }
```

Comment:
Instead of using raw layout, this probably should dispatch to `merkle_root` methods:
https://github.com/solana-labs/solana/blob/a9509f56b/ledger/src/shred/merkle.rs#L157
https://github.com/solana-labs/solana/blob/a9509f56b/ledger/src/shred/merkle.rs#L269

Reply:
noted, will do this in the following PR